chat-android/src/main/java/com/ably/chat/ErrorCodes.kt

```diff
@@ -3,105 +3,106 @@ package com.ably.chat
 /**
  * Error codes for the Chat SDK.
  */
-object ErrorCodes {
```

Comment:
Why do you think enum here is better?

Reply:
Having type for given `ErrorCode` enforces compile time check with error code reusability. 
Also, IDE doesn't give right suggestions when matching error codes when it's generic `Int`.
This makes it difficult to write correct assertions in code as well as tests.

Reply:
Also, in chat-js, [`ErrorCode` is represented as an enum](https://github.com/ably/ably-chat-js/blob/fccf81494b998d8977c88c9cac09dac9b3eb8c16/src/core/errors.ts#L6), this makes it super easy to correlate between two codebases.

Reply:
Not sure I understand, could you give an example? Typescript's enums are nothing like Kotlin's enums. Error codes usually come from backend, and they come as `Int`. I currently don't understand what benefits we get, except we have to additionally serialize/deserialize `Int` into enum

Reply:
Sure, I am closing this PR in favor of https://github.com/ably-labs/ably-chat-kotlin/pull/33. We can continue discussion for the same there 👍